tablet_integration_test.go

```diff
@@ -0,0 +1,157 @@
+//go:build tablet
+// +build tablet
+
+package gocql
+
+import (
+	"bytes"
+	"context"
+	"fmt"
+	"regexp"
+	"strings"
+	"testing"
+	"time"
+)
+
+// Check if TokenAwareHostPolicy works correctly when using tablets
+func TestTablets(t *testing.T) {
+	cluster := createMultiNodeCluster()
+	cluster.EnableExperimentalTablets()
+
+	fallback := RoundRobinHostPolicy()
+	cluster.PoolConfig.HostSelectionPolicy = TokenAwareHostPolicy(fallback)
+
+	session := createSessionFromMultiNodeCluster(cluster, t)
+	defer session.Close()
+
+	if err := createTable(session, fmt.Sprintf(`CREATE TABLE %s.%s (pk int, ck int, v int, PRIMARY KEY (pk, ck));
+	`, "test1", "table1")); err != nil {
+		panic(fmt.Sprintf("unable to create table: %v", err))
+	}
+
+	hosts, _, err := session.hostSource.GetHosts()
+	assertTrue(t, "err == nil", err == nil)
+
+	hostAddresses := []string{}
+	for _, host := range hosts {
+		hostAddresses = append(hostAddresses, host.connectAddress.String())
+	}
+
+	ctx := context.Background()
+
+	i := 0
+	for i < 50 {
+		i = i + 1
+		err = session.Query(`INSERT INTO test1.table1 (pk, ck, v) VALUES (?, ?, ?);`, i, i%5, i%2).WithContext(ctx).RoutingKey([]byte{}).Exec()
+		assertTrue(t, "err == nil", err == nil)
+	}
+
+	time.Sleep(1 * time.Minute)
+
+	i = 0
+	for i < 50 {
+		i = i + 1
+
+		var pk int
+		var ck int
+		var v int
+
+		buf := &bytes.Buffer{}
+		trace := NewTraceWriter(session, buf)
+
+		err = session.Query(`SELECT pk, ck, v FROM test1.table1 WHERE pk = ?;`, i).WithContext(ctx).Consistency(One).Trace(trace).Scan(&pk, &ck, &v)
+		assertTrue(t, "err == nil", err == nil)
+
+		queriedHosts := 0
+		for _, hostAddress := range hostAddresses {
+			if strings.Contains(buf.String(), hostAddress) {
+				queriedHosts = queriedHosts + 1
+			}
+		}
+
+		assertEqual(t, "queriedHosts", 1, queriedHosts)
```

Comment:
Please add a (temporary?) print here of all `hostAddresses` (before `strings.Contains` filter) so you can debug the failing CI.